ez/ez.go

```diff
@@ -69,6 +76,23 @@ func WithOnWatchedError(cb dials.WatchedErrorHandler) Option {
 	return func(d *dialsOptions) { d.onWatchedError = cb }
 }
 
+// WithFileKeyNaming provides a method for manipulating the casing of the keys
+// in the configuration file.  The decoder argument indicates how to interpret
+// the `dials` tag's formatting.  If the `dials` tag is unspecified, the struct
+// field.  The encoder argument should indicate the format that dials should
+// expect to find in the file.  For instance if you leave the `dials` tag
+// unspecified and want a field named `SecretValues` in your configuration to
+// map to a value in your config named "secret-values" you should call:
+// WithFileKeyNaming(caseconversion.DecodeGoCamelCase,
+// caseconversion.EncodeKebabCase). Note that this does not affect the flags or
+// environment.
```

Comment:
Let's add a parenthetical mentioning that a for flags, they'd want `WithFlagConfig` to adjust naming.